controllers/controllers/client.go

```diff
@@ -0,0 +1,196 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+
+	anywhere "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	spec "github.com/aws/eks-anywhere/pkg/cluster"
+	anywhererelease "github.com/aws/eks-anywhere/release/api/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capicontrolplane "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *ClusterReconcilerV2) BuildRemoteClientForCluster(ctx context.Context, cluster *anywhere.Cluster) (*eksaClient, error) {
+	capiCluster, err := r.Client.GetCAPICluster(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+	key := client.ObjectKey{
+		Namespace: capiCluster.Namespace,
+		Name:      capiCluster.Name,
+	}
+	remoteClient, err := r.Tracker.GetClient(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	return &eksaClient{Client: remoteClient}, nil
+}
+
+type eksaClient struct {
+	client.Client
+}
+
+func (c eksaClient) BuildClusterSpec(ctx context.Context, cluster *anywhere.Cluster) (*cluster.Spec, error) {
+	bundles, err := c.GetBundlesForCluster(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return spec.BuildSpecFromBundles(cluster, bundles)
+}
+
+func (c eksaClient) GetBundlesForCluster(ctx context.Context, cluster *anywhere.Cluster) (*anywhererelease.Bundles, error) {
+	// TODO: the bundles name can be cached in the status
+	// TODO: answer this question: do the management cluster and all the workload clusters use the same Bundle?
+	//  if they don't, that opens a lot our testing matrix. Right now they do and they don't. The cli is not enforcing it,
+	//  but the v0.6.0 is the only cli that supports workload clusters and there is only one bundle compatible with it
+	// For now, I'm gonna get the bundles using the cluster name, but we should really change that
+
+	// Once we have the bundles ref in the status, we can add a check here and use it if present
+	// If the status is not there, it means it's a new cluster, so we get the bundles being used by the management cluster and use it
+	// Maybe using the spec would be better than the status. Status is supposed to be recoverable from the state of the cluster,
```

Comment:
+1 to spec. user can specify bundleRef and overrides with default too. this will replace the `--bundles-override` in CLI if i understand correct.